src/AuthContext.tsx

```diff
@@ -109,8 +122,10 @@ export const AuthProvider = ({ authConfig, children }: IAuthProvider) => {
             }
           })
       } else {
-        // The refresh token has expired. Need to log in from scratch.
-        login()
+        if (initial) return login()
+
+        if (onRefreshTokenExpire) onRefreshTokenExpire({ login } as TRefreshTokenExpiredEvent)
+        else login() // TODO Breaking change - remove automatic login during ongoing session
```

Comment:
This isn't really how I would've preferred to solve this problem, but as this will only be a minor-upgrade, there can be no breaking changes. For the next major version, I suggest updating these lines to: 
```typescript
if (initial) {
  if (autoLogin) login()
  if (!autoLogin) clearStorage()
  return
}

if (onRefreshTokenExpire) onRefreshTokenExpire({ login } as TRefreshTokenExpiredEvent)
else {
  setError('Access and refresh tokens expired.')
  console.warn('Access and refresh tokens expired.')
}
```

The current behaviour could be considered a bug, but at the same time it is now a feature.